tests/components/devolo_home_control/test_binary_sensor.py

```diff
@@ -0,0 +1,109 @@
+    assert state.state == STATE_OFF
+
+    # Emulate websocket message: sensor turned on
+    HomeControlMockBinarySensor.publisher.dispatch("Test", ("Test", True))
+    await hass.async_block_till_done()
+    assert hass.states.get(f"{DOMAIN}.test").state == STATE_ON
+
+    # Emulate websocket message: device went offline
+    DeviceMock.available = False
+    HomeControlMockBinarySensor.publisher.dispatch("Test", ("Status", False, "status"))
+    await hass.async_block_till_done()
+    assert hass.states.get(f"{DOMAIN}.test").state == STATE_UNAVAILABLE
+
+
+@pytest.mark.usefixtures("mock_zeroconf")
+async def test_remote_control(hass: HomeAssistant):
+    """Test setup and state change of a remote control device."""
+    entry = configure_integration(hass)
+    DeviceMock.available = True
+    with patch(
+        "homeassistant.components.devolo_home_control.HomeControl",
+        side_effect=[HomeControlMockRemoteControl, HomeControlMock],
+    ):
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+    state = hass.states.get(f"{DOMAIN}.test")
+    assert state is not None
+    assert state.state == STATE_OFF
+
+    # Emulate websocket message: button pressed
+    HomeControlMockRemoteControl.publisher.dispatch("Test", ("Test", 1))
+    await hass.async_block_till_done()
+    assert hass.states.get(f"{DOMAIN}.test").state == STATE_ON
+
+    # Emulate websocket message: button released
+    HomeControlMockRemoteControl.publisher.dispatch("Test", ("Test", 0))
+    await hass.async_block_till_done()
+    assert hass.states.get(f"{DOMAIN}.test").state == STATE_OFF
+
+    # Emulate websocket message: device went offline
+    DeviceMock.available = False
+    HomeControlMockRemoteControl.publisher.dispatch("Test", ("Status", False, "status"))
+    await hass.async_block_till_done()
+    assert hass.states.get(f"{DOMAIN}.test").state == STATE_UNAVAILABLE
+
+
+@pytest.mark.usefixtures("mock_zeroconf")
+async def test_disabled(hass: HomeAssistant):
+    """Test setup of a disabled device."""
+    entry = configure_integration(hass)
+    with patch(
+        "homeassistant.components.devolo_home_control.HomeControl",
+        side_effect=[HomeControlMockDisabledBinarySensor, HomeControlMock],
+    ):
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert hass.states.get(f"{DOMAIN}.devolo.WarningBinaryFI:Test") is None
+
+
+@pytest.mark.usefixtures("mock_zeroconf")
+async def test_remove_from_hass(hass: HomeAssistant):
+    """Test removing entity."""
+    entry = configure_integration(hass)
+    with patch(
+        "homeassistant.components.devolo_home_control.HomeControl",
+        side_effect=[HomeControlMockBinarySensor, HomeControlMock],
+    ):
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+        await hass.config_entries.async_remove(entry.entry_id)
```

Comment:
Maybe assert that there is a state in the state machine for our integration before removing the config entry?